### PR TITLE
Search reaction by name/keywords

### PR DESCRIPTION
**Pull request by BillCarsonFr**
searchEmoji

### Pull Request Checklist

- [ ] UI change has been tested on both light and dark themes
- [ ] Pull request is based on the develop branch
- [ ] Pull request updates [CHANGES.md](https://github.com/vector-im/riotX-android/blob/develop/CHANGES.md)
- [ ] Pull request includes screenshots or videos if containing UI changes

### CODE REVIEW

File: vector/src/main/java/im/vector/riotx/features/reactions/EmojiChooserFragment.kt [hunk 1]
```diff
@@ -36,12 +36,10 @@ class EmojiChooserFragment : VectorBaseFragment() {
         viewModel = activity?.run {
             ViewModelProviders.of(this, viewModelFactory).get(EmojiChooserViewModel::class.java)
         } ?: throw Exception("Invalid Activity")
-        viewModel.initWithContect(context!!)
+        viewModel.initWithContext(context!!)
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotx/features/reactions/EmojiChooserFragment.kt]: Don't pass activity context to viewModel. Please use injection to pick what you need from Application context

File: vector/src/main/java/im/vector/riotx/features/reactions/EmojiReactionPickerActivity.kt [hunk 2]
```diff
@@ -150,10 +164,34 @@ class EmojiReactionPickerActivity : VectorBaseActivity(), EmojiCompatFontProvide
...
                     findViewById<Toolbar>(R.id.toolbar)?.minimumHeight = 0
-                    it.setQuery("", true)
+                    searchView.setQuery("", true)
                     return true
                 }
             })
+
+            val searchObservable = Observable.create<String> { emitter ->
+
+                searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotx/features/reactions/EmojiReactionPickerActivity.kt]: Can't work with RxBinding as discussed IRL?

File: vector/src/main/java/im/vector/riotx/features/reactions/EmojiSearchResultController.kt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import javax.inject.Inject
+
+class EmojiSearchResultController @Inject constructor(val stringProvider: StringProvider,
+                                                      fontProvider: EmojiCompatFontProvider)
+    : TypedEpoxyController<EmojiSearchResultViewState>() {
+
+    var emojiTypeface: Typeface? = fontProvider.typeface
+
+    init {
+        fontProvider.addListener(object : EmojiCompatFontProvider.FontProviderListener {
```
Comment (ganfra) [inline vector/src/main/java/im/vector/riotx/features/reactions/EmojiSearchResultController.kt]: You add listeners you never remove on a singleton instance :/

File: vector/src/main/java/im/vector/riotx/features/reactions/EmojiChooserFragment.kt (hunk 1 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/riotx/features/reactions/EmojiChooserFragment.kt]: Yes, this fragment need to be refactored completely. Maybe another PR?

File: vector/src/main/java/im/vector/riotx/features/reactions/EmojiReactionPickerActivity.kt (hunk 2 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/riotx/features/reactions/EmojiReactionPickerActivity.kt]: done

File: vector/src/main/java/im/vector/riotx/features/reactions/EmojiSearchResultController.kt (hunk 3 above)
Comment (BillCarsonFr) [reply vector/src/main/java/im/vector/riotx/features/reactions/EmojiSearchResultController.kt]: done

**Pull request merged**